app/lib/transitions/view-transition.tsx

```diff
@@ -216,7 +216,7 @@ export function LinkWithViewTransition<
     prefetch: props.prefetch ?? 'viewport',
     onClick: props.onClick,
     viewTransition: true,
-    state: linkState,
+    state: { ...props.state, ...linkState },
```

Comment:
this is completely unrelated to this approach. while exploring one other approach before I noticed that if you pass state to our link component it gets lost. this fixes it